tiup: add command docs of clean, install and status (#4979)
pingcap/docs: tiup/tiup-command-install.md

---
title: tiup install
---

# tiup install

> The command `tiup install` is used for component installation. It downloads the component package of the specified version from the mirror repository and decompresses it in the local TiUP data directory for later use. In addition, when TiUP needs to run a component that does not exist in the mirror warehouse, it tries to download the component first and then runs it automatically. If it does not exist in the warehouse, an error is reported.

Reviewer: ```suggestion
The command `tiup install` is used for component installation. It downloads the component package of a specified version from the mirror repository and decompresses it in the local TiUP data directory for later use. In addition, when TiUP needs to run a component that does not exist in the mirror repository, it tries to download the component first and then runs it automatically. If the component does not exist in the repository, an error is reported.
```

## Syntax

```sh
tiup install <component1>[:version] [component2...N] [flags]
```

`<component1>` and `<component2>` represent component names, and `[version]` represents an optional version number. If `version` is not added, the latest stable version of the specified component is installed. `[component2...N]` means that you can specify multiple components or multiple versions of the same component at the same time.

## Option

None

## Output

> - Under normal circumstances, output the download information of components.

Reviewer: ```suggestion
- Normally outputs the download information of the component.
```

> - If the component does not exist, an error `The component "%s" not found` is reported.

Reviewer: ```suggestion
- If the component does not exist, the `The component "%s" not found` error is reported.
```

> - If the version does not exist, an error `version %s not supported by component %s` is reported.

Reviewer: ```suggestion
- If the version does not exist, the `version %s not supported by component %s` error is reported.
```
